tests/pfcwd/test_pfcwd_warm_reboot.py

```diff
@@ -0,0 +1,470 @@
+
+        Args:
+            testcase_actions(list): list of actions that the test will go through
+            setup_pfc_test(fixture): module scoped autouse fixture
+            fanout_graph_facts(fixture): fanout info
+            ptfhost(AnsibleHost): PTF instance
+            duthost(AnsibleHost): DUT instance
+            localhost(AnsibleHost): local instance
+            fanouthosts(AnsibleHost): fanout instance
+        """
+        setup_info = setup_pfc_test
+        self.fanout_info = fanout_graph_facts
+        self.ptf = ptfhost
+        self.dut = duthost
+        self.fanout = fanouthosts
+        self.timers = setup_info['pfc_timers']
+        self.ports = setup_info['selected_test_ports']
+        self.neighbors = setup_info['neighbors']
+        dut_facts = self.dut.setup()['ansible_facts']
+        self.peer_dev_list = dict()
+        self.seed = int(datetime.datetime.today().day)
+        self.storm_handle = dict()
+        bitmask = 0
+        storm_deferred = 0
+        storm_restored = 0
+        self.max_wait = 0
+
+        for t_idx, test_action in enumerate(testcase_actions):
+            if 'warm-reboot' in test_action:
+                reboot(self.dut, localhost, reboot_type="warm")
+                continue
+
+            # one of the factors to decide if the storm needs to be started
+            storm_restored = bitmask and (bitmask & 2)
+            # if the action prior to the warm-reboot was a 'storm_defer', ensure that all the storms are
+            # stopped
+            storm_deferred = bitmask and (bitmask & 4)
+            if storm_deferred:
+                logger.info("Wait for all the deferred storms to start and stop ...")
+                time.sleep(self.max_wait)
+                self.storm_handle = dict()
+
+            bitmask = (1 << ACTIONS[test_action])
+            for p_idx, port in enumerate(self.ports):
+                 logger.info("")
+                 logger.info("--- Testing on {} ---".format(port))
+                 self.setup_test_params(port, setup_info['vlan'])
+                 for q_idx, queue in enumerate(self.pfc_wd['queue_indices']):
+                     if not t_idx or storm_deferred:
+                         if not q_idx:
+                             self.storm_handle[port] = dict()
+                         self.storm_setup(port, queue, storm_defer=(bitmask & 4))
+
+                     self.traffic_inst = SendVerifyTraffic(self.ptf, dut_facts['ansible_eth0']['macaddress'], self.pfc_wd, queue)
+                     self.run_test(port, queue, detect=(bitmask & 1),
+                                   storm_start=not t_idx or storm_deferred or storm_restored,
+                                   first_detect_after_wb=(t_idx == 2 and not p_idx and not q_idx and not storm_deferred),
+                                   storm_defer=(bitmask & 4))
+
+    def test_pfcwd_no_storm_wb(self, setup_pfc_test, fanout_graph_facts, ptfhost, duthost, localhost, fanouthosts):
```

Comment:
I feel this test case and subsequent one could be parameterized by TESTCASE_ACTION